### DISCUSSION TITLE
Is there any functions that I can arbitrarly control the acceleration of a drone?

### DISCUSSION THREAD

**Discussion by alkimo13** [USC-ACTLab/crazyswarm#706] [Q&A]

**Comment by whoenig** [comment]
You could use the FullStateCommand for online operation or trajectories for offline planning. In both cases, you will also need to provide velocities and positions for the existing on-board controllers.

**Reply by alkimo13** [reply]
1. Are cmdFullState and FullStateCommand same things?
2. If we are providing a velocity parameter, what a position parameter means? it only works if I update the position parameter with respect to velocity and give it back to the function. For example I can decrease or increase the value of velocity in a loop manually while I am entering (0,0,0) for the acceleration parameter of the function. What am I missing here?

**Reply by whoenig** [reply]
1. Yes
2. This depends on the controller. The existing controllers try to track position and velocity and use the acceleration as feed-forward term. The firmware also has a mode for just velocity control (not sure if this is currently exposed in Crazyswarm1), but to my knowledge no mode for acceleration control (since there is no "clean" feedback signal anyways.)
